server/src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -649,10 +649,10 @@ private IndexShardState changeState(IndexShardState newState, String reason) {
             isRetry, Engine.Operation.Origin.PRIMARY, sourceToParse);
     }
 
-    public Engine.IndexResult applyIndexOperationOnReplica(long seqNo, long version, VersionType versionType,
-                                                           long autoGeneratedTimeStamp, boolean isRetry, SourceToParse sourceToParse)
+    public Engine.IndexResult applyIndexOperationOnReplica(long seqNo, long version, long autoGeneratedTimeStamp,
+                                                           boolean isRetry, SourceToParse sourceToParse)
         throws IOException {
-        return applyIndexOperation(seqNo, primaryTerm, version, versionType, autoGeneratedTimeStamp, isRetry,
+        return applyIndexOperation(seqNo, primaryTerm, version, VersionType.EXTERNAL, autoGeneratedTimeStamp, isRetry,
```

Comment:
maybe it's better to pass null instead of VersionType.EXTERNAL? this way people may think it's the original one?

Reply:
if we do that, maybe assert in the Engine.Op constructor that if the origin is not a primary, version type is null?

Reply:
+1 on this suggestion